src/builders/indexable-builder.php

```diff
@@ -299,6 +299,29 @@ protected function save_indexable( $indexable, $indexable_before = null ) {
 		return $indexable;
 	}
 
+	/**
+	 * Build and author indexable from an author id if it does not exist yet.
+	 *
+	 * @param int $author_id The author id.
+	 *
+	 * @return false|Indexable
+	 */
+	protected function maybe_build_author_indexable( $author_id ) {
+		$author_indexable = $this->indexable_repository->find_by_id_and_type(
+			$author_id,
+			'user',
+			false
+		);
+		if ( ! $author_indexable || $this->version_manager->indexable_needs_upgrade( $author_indexable ) ) {
```

Comment:
```suggestion
		if ( ! $author_indexable ) {
```
I don't think we need this check, since the `$this->indexable_repository->find_by_id_and_type` call we do above already does an upgrade when needed 🤔 . 

Changing it may be out of scope if this issue, though, since it is a pre-existing thing.

Reply:
I ultimately decided to not change this because:

- It is a pre-existing issue, copied over from the original implementation.
- I'm not sure about the impact of removing this.
- Checking this twice does not hurt, it does not have a negative impact on performance.